app/scripts/test-helpers/test-helpers.jsx

```diff
@@ -241,6 +291,132 @@ export const mountHGComponent = (
+    // stopped changing sizes
+    maxTime = 1000;
+  }
+
+  // Initialize to empty elements
+  const prevSizes = elements.map((element) => ({
+    width: 0,
+    height: 0,
+  }));
+
+  for (let i = 0; i < maxTime; i += timeInterval) {
+    // Every time interval, get their new size
+    const newSizes = elements.map((element) => ({
+      width: element.getBoundingClientRect().width,
+      height: element.getBoundingClientRect().height,
+    }));
+
+    let different = false;
+
+    // And check if any have changed size
+    for (let j = 0; j < newSizes.length; j++) {
+      if (
+        newSizes[j].width !== prevSizes[j].width ||
+        newSizes[j].height !== prevSizes[j].height
+      ) {
+        // Something has changed size, make a note of it
+        // and store the new sizes to compare in the next
+        // iteration
+        different = true;
+        prevSizes[j].width = newSizes[j].width;
+        prevSizes[j].height = newSizes[j].height;
+      }
+    }
+
+    // If nothing has changed, we're done
+    if (!different) return;
+
+    // Wait until the next check
+    await new Promise((r) => setTimeout(r, timeInterval));
+  }
+};
+
+/** Wait for scales to stop changing.
+ *
+ * @param {HiGlassComponent} hgc
+ * @param {string} viewUid
+ * @param {number} [timeInterval] - The interval (in milliseconds) between size checks.
+ * @param {number} [maxTime] - The maximum time (in milliseconds) to wait for stabilization.
+ * @returns
+ */
+export const waitForScalesStabilized = async (
+  hgc,
+  viewUid,
+  timeInterval = 50,
+  maxTime = 2000,
+) => {
+  const xScaleDomain = [0, 0];
+  const yScaleDomain = [0, 0];
+
+  for (let i = 0; i < maxTime; i += timeInterval) {
+    const xScale = hgc.xScales[viewUid];
+    const yScale = hgc.yScales[viewUid];
+
+    console.log(
+      'xScale.domain()',
+      xScale.domain(),
+      'yScale.domain()',
+      yScale.domain(),
+    );
+    if (
+      xScaleDomain[0] !== xScale.domain()[0] ||
+      xScaleDomain[1] !== xScale.domain()[1] ||
+      yScaleDomain[0] !== yScale.domain()[0] ||
+      yScaleDomain[1] !== yScale.domain()[1]
+    ) {
+      xScaleDomain[0] = xScale.domain()[0];
+      xScaleDomain[1] = xScale.domain()[1];
+      yScaleDomain[0] = yScale.domain()[0];
+      yScaleDomain[1] = yScale.domain()[1];
+    } else {
+      return;
+    }
+
+    await new Promise((r) => setTimeout(r, timeInterval));
+  }
+};
+
+/**
+ * Wait for a HiGlassComponet to be ready at the given element.
+ *
+ * By ready we mean that a track-renderer-div is present and that its
+ * size is not changing any more.
+ *
+ * @param {HTMLElement} div
+ */
+export const waitForComponentReady = async (div) => {
+  const elementQueries = ['.track-renderer-div'];
+  await new Promise((r) => waitForElements(div, elementQueries, r));
+
+  await waitForSizeStabilization(
+    // Check for size changes every 20 ms for 2000 seconds
+    elementQueries.map((x) => div.querySelector(x), 20, 3000),
+  );
```

Comment:
Think this is a bug.

```suggestion
  await waitForSizeStabilization(
    // Check for size changes every 20 ms for 2000 seconds
    elementQueries.map((x) => div.querySelector(x)),
    20,
    3000,
  );
```

Reply:
I opened https://github.com/higlass/higlass/pull/1254 against this branch which types `test-helpers.jsx` and enables type-checking so there is better auto complete, static analysis of the code here.

Reply:
Thanks! I just pushed some more changes and am hoping the tests finally pass (they pass locally) and then we can merge that branch in?

Reply:
> merge that branch in?

sounds good to me, I can try to rebase the changes...

Reply:
I rebased the changes on #1254